src/Microsoft.AspNetCore.Rewrite/Internal/Shared/PatternSegments/IsHttp2Segment.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Rewrite.Internal.Shared.PatternSegments
+{
+    public class IsHttp2Segment : PatternSegment
+    {
+        public override string Evaluate(HttpContext context, MatchResults ruleMatch, MatchResults condMatch)
+        {
+            return context.Request.Scheme == "http2" ? "on" : "off";
```

Comment:
This is not how to detect http/2. We don't surface this information anywhere at the moment so you'll have to disable this feature.